UI/js/controllers.js

```diff
@@ -126,35 +126,40 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
   //use the Terms factory as a promise. Add returned data to the scope
 
   Terms.getTerms(termsUrl).then(function (data) {
-    // the ESB might return a sinble object rather than an array, turn it into an array
-    if (data.Result.length === undefined ){
-      data.Result = [].concat(data.Result);
+    if (data.failure) {
+      $scope.$parent.term  = $rootScope.lang.termFailure
     }
-    if (data.Result.length !==0)  {
+    else {
+      // the ESB might return a sinble object rather than an array, turn it into an array
+      if (data.Result.length === undefined ){
+        data.Result = [].concat(data.Result);
+      }
+      if (data.Result.length !==0)  {
 
-      $scope.terms = data.Result;
-      $scope.$parent.term = data.Result[0].TermDescr;
-      $scope.$parent.termId = data.Result[0].TermCode;
+        $scope.terms = data.Result;
+        $scope.$parent.term = data.Result[0].TermDescr;
```

Comment:
So - the cascade goes something like this:
error >  display message  
no error > data is object > turn into an array with one object
no error > data is array > first term is the current term